File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public static int first(int[] nums, int target){
        int start = 0 , end = nums.length-1;
        int ans = -1;
        while(start<=end){
            int mid = (start+end)/2;
            if(nums[mid]==target){
                ans = mid;
                end = mid-1;
            }
            else if(nums[mid]> target){
                end = mid-1;
            }
            else start = mid+1;
        }
        return ans;
    }
                    
    public static int last(int[] nums, int target){
        int start = 0 , end = nums.length-1;
        int ans = -1;
        while(start<=end){
            int mid = (start+end)/2;
            if(nums[mid]==target){
                ans = mid;
                start=mid+1;
            }
            else if(nums[mid]> target){
                end = mid-1;
            }
            else start = mid+1;
        }
        return ans;
    }
    public int[] searchRange(int[] nums, int target) {
        int num1 = first(nums,target);
        int num2 = last(nums,target);
        int [] ans = {num1, num2};
        return ans;
        
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    int first_ocurr(vector<int> & nums, int target){
        int n=nums.size();
        int l=0,h=n-1;
        int ans=-1;
        while(l<=h){
            int mid= l + (h-l)/2;
            if(nums[mid]==target){
                ans=mid;
                h=mid-1;
            }
            else if(nums[mid]>target)h=mid-1;
            else{
                l=mid+1;
            }
        }
        
        return ans;
    }
    
    int last_ocurr(vector<int> & nums, int target){
        int n=nums.size();
        int l=0,h=n-1;
        int ans=-1;
        while(l<=h){
            int mid= l + (h-l)/2;
            if(nums[mid]==target){
                ans=mid;
                l=mid+1;
            }
            else if(nums[mid]<target)l=mid+1;
            else{
                h=mid-1;
            }
        }
        
        return ans;
    }
    
    
    
    vector<int> searchRange(vector<int>& nums, int target) {
        int a=first_ocurr(nums,target);
        int b=last_ocurr(nums,target);
        vector<int> ans={a,b};
        return ans;
    }
};